misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -231,6 +253,71 @@ internal class SqsJobConsumer @Inject internal constructor(
         )
       }
     }
+
+  }
+
+  internal inner class BatchQueueReceiver(
+    queueName: QueueName,
+    private val handler: BatchJobHandler
+  ) : QueueReceiver(queueName) {
+
+    override fun receive(): List<Status> {
+      // Repeatedly long poll for messages until we have batchSize messages.
+      // If there are no messages available, short circuit further requests.
+      val cutoff = TimeSource.Monotonic.markNow() + 20.seconds // TODO make configurable
+      val batch: MutableList<SqsJob> = mutableListOf()
+      while (batch.size < batchSize() && !cutoff.hasPassedNow()) {
+        val messages = fetchMessages(
+          sqsBatchSize = min(10, batchSize() - batch.size),
+          waitTimeSeconds = 5
+        )
+        batch.addAll(messages)
+        if (messages.isEmpty()) {
+          break
+        }
+      }
+
+      if (batch.isEmpty()) {
+        return listOf(Status.NO_WORK)
+      }
+
+      return listOf(handleMessages(batch))
+    }
+
+    @OptIn(ExperimentalMiskApi::class)
+    private fun handleMessages(messages: List<SqsJob>): Status {
+      metrics.jobsReceived.labels(queue.queueName, queue.queueName).inc(messages.size.toDouble())
+
+      return tracer.traceWithNewRootSpan("handle-job-${queue.queueName}") { span ->
+        // Run the handler and record timing
+        try {
+          messages.forEachIndexed { index, message ->
+            MDC.put("$SQS_JOB_IDS_STRUCTURED_MDC.$index", message.id)
```

Comment:
How can I log a list of IDs? DD supports lists in log attributes but I don't know what key format to use.